src/tools/rbd_mirror/leader_watcher/Types.h

```diff
@@ -68,6 +136,10 @@ struct UnknownPayload {
 typedef boost::variant<HeartbeatPayload,
                        LockAcquiredPayload,
                        LockReleasedPayload,
+                       SyncRequestPayload,
+                       SyncRequestAckPayload,
```

Comment:
Nit: perhaps this message could be called ```SyncStartPayload``` (from leader -> instance to grant permission to start the sync) and we could eliminate the previous use of ```SyncStartPayload``` below. The ```Ack``` suffix just seems odd to me since it's not an acknowledgement of the request but rather a command to start.